components/size_aware_image.jsx

```diff
@@ -148,10 +147,22 @@ export default class SizeAwareImage extends React.PureComponent {
                         />
                     </div>
                 }
-                <img
-                    {...props}
-                    src={src}
-                />
+                {dimensions && dimensions.width && !this.state.loaded ? (
```

Comment:
Just a thought, but because this condition is fairly complicated, it might make sense to move this out to an if block instead of using a ternary statement in the JSX. 0/5 on that though.

Reply:
Makes sense. Moved them into instance functions.